### ISSUE TITLE
macOS - disable resize control in lower right corner?

### ISSUE THREAD

**Issue by emendelson** [joncampbell123/dosbox-x#642]
When I run DOSBox-X in macOS, a resizing control appears at the lower right of the window, like this:

![screen shot 2018-04-01 at 8 46 08 am](https://user-images.githubusercontent.com/5208505/38173270-61347dbc-3589-11e8-8fc1-94391bdccf09.png)

I see that I can get rid of the control by changing the output from surface to overlay, but that changes causes other problems that I'm trying to avoid. Is there a config setting that prevents this control from appearing with output=surface?

**Comment by joncampbell123** [comment]
I'll add an option to dosbox.conf to enable/disable resizing.

The display contents should stretch to fill the window if output=opengl.

**Comment by emendelson** [comment]
This is a strictly selfish request, and please feel free to ignore it without apologies. The current code is working so well that I would like to switch all my active DOSBox-based projects on Windows and the Mac from SVN to DOSBox-X. The only thing that gets in my way is the ability to resize an active DOSBox-X window, especially on the Mac, where there's a distracting resize corner tab at the lower right of the window. 

I've added a mapper key that runs GUI_ResetResize, but that tab is still distracting and many of my low-tech users won't know how to use the mapper key.

Could you point me in the direction of where I might experiment with disabling window resize? All I would need to do for my project is disable resizing - and so there's no hurry about a dosbox.conf setting. Again, I'm not asking you to do the work for me - just to point me in the right direction, if possible. Thank you!

**Comment by joncampbell123** [comment]
Look for calls to SDL_SetVideoMode and where it might set the SDL_RESIZABLE flag on the call.

Removing SDL_RESIZEABLE should get rid of the resize box.

**Comment by emendelson** [comment]
And that change in sdlmain.cpp line 1297 removes the resize tab in MacOS. Thank you again.

**Comment by emendelson** [comment]
I hope it's OK to renew the request for a dosbox.conf option to enable/disable resizing? I've been modifying the sdlmain.cpp code as it changes, but I think I'm not the only person would would be glad to be able to turn off resizing - especially in text screens, which look very strange when stretched. Thank you!

**Comment by aybe** [comment]
@emendelson 

I believe this should be done differently to retain a consistent behavior in the host OS,

In Windows, users are accustomed to be able to resize a normal window (with borders), a non-resize-able window such as a tool window has no borders, which makes it instantly obvious that it is not sizable. I really don't see the point of having a normal (resize-able) window to not be resize-able.

![ic93227](https://user-images.githubusercontent.com/1537591/42065821-8f0d82a2-7b3d-11e8-8981-26a3c85237ce.png)

https://msdn.microsoft.com/en-us/library/system.windows.window.windowstyle%28v=vs.110%29.aspx

What I suggest instead would be to add a 'snap to integer scale when resizing' feature.

And stretching is just too useful to be removed in my opinion, sometimes you want to maximize screen space regardless visual impact.

TL;DR; DOSBox-X **normal** window should not behave differently than any other application.

Alternatively, don't resize manually or apply a scaler in the menu to size it to what you'd like.

@joncampbell123 

Your opinion on this ?

**Comment by emendelson** [comment]
This is very intriguing. The big difference between a DOSBox-X window and most (not all) non-DOSBox windows is that the **contents** of the non-DOSBox windows (the text and graphics) stays the same size when you stretch the window border. So if I stretch the border of the browser window that I'm typing in now, the text and graphics always remain the same size. 

I certainly see the point of stretching a DOSBox-X window, but the trouble in the current code is that you can stretch the window to different vertical and horizontal percentages, so that the text in a window might

s t a r t   t o   l o o k   l i k e   t h i s

and a 9x14 (or whatever) character cell might become a 9x28 character cell.

Could stretching be made to preserve aspect ratio? Is that what you mean by "snap to integer scale"? I can't really imagine a situation where you would want to stretch a game horizontally but not vertically when it's in a window, but maybe I haven't thought it through. (I can see that unequal stretching might make sense in full-screen mode, but not in windowed mode.)

Am I getting all this wrong?

**Comment by aybe** [comment]
I see what you mean, I believe this may come from rounding issues in the current scaler (didn't check though).

What I mean,

Stretching: makes it exactly how big the user wants it, at the expense of poor/blurry rendering and/or weird aspect

Snap integer scaling on the other hand would preserve the aspect ratio and since it snaps, expect pixel-perfect rendering.

So by having these two modes, everyone is happy, no matter what their expectations are.

**Comment by joncampbell123** [comment]
@aybe Exactly. If the user doesn't want to preserve aspect ratio, then the user should be able to stretch and distort the screen however he/she wants.

**Comment by emendelson** [comment]
That makes perfect sense: snap-integer-scaling would prevent ugly distortions and it would be easy to recover from a mistaken stretch.

**Comment by emendelson** [comment]
Apologies for bringing this up again. During the summer, I was able to prevent resizing the window by adding some SDL_RESIZABLE(0) strings to sdlmain.cpp. When I tried this with current code, it seems to have had no effect - I can resize the window, which is something I don't want to do in DOS text-based applications that have fixed lines-columns setting.

Is there a current way to prevent window-resizing? The users I support will only be confused when they resize by accident, and there's no obvious way to get back to 80x25 text once you drag the borders.

Thank you again for any help with this!

**Comment by joncampbell123** [comment]
You may not have put #define SDL_RESIZEABLE (0) in every source file that uses SDL_RESIZEABLE.

At the same time, there is a "reset window size" command in the menu to restore the normal window size.

**Comment by emendelson** [comment]
That's exactly what I neglected to do. Thank you. Actually, I was able to get the same result by changing the definition of SDL_RESIZABLE in sdl_video.h from 1 to 0, but I'll try to find a way that uses the DOSBox-X code so that I can add a config setting for resizable or not.

Thank you again!
